File: Python/Leetcode/HashMap/ransomNote.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        ranMap = {}
        for char in ransomNote:
            ranMap[char] = ranMap.get(char, 0) + 1

        for char in magazine:
            ranMap[char] = ranMap.get(char, 0) - 1
        
        for i, val in ranMap.items():
            if val > 0:
                return False
        return True
    
if __name__ == '__main__':
    solution_instance = Solution()
    solution_instance.canConstruct("aa", "aab")

File: C++/Leetcode/Hashmap/ransomNote.cpp
#include <iostream>
#include <map>
#include <string>

class Solution{
    public:
        bool canConstruct(std::string ransomNote, std::string magazine){
            std::map<char, int> rMap;
            for (const auto& c : ransomNote){
                rMap[c]++;
            }

            for (const auto& c : magazine){
                rMap[c]--;
            }

            for (std::map<char, int>::iterator itr = rMap.begin(); itr != rMap.end(); ++itr){
                if (itr->second > 0){
                    return false;
                }
            }
            return true;
        }
};
                                         
int main(){
    Solution solution_instance;
    std::cout << solution_instance.canConstruct("aa", "aab") << std::endl;
}